client.go

```diff
@@ -0,0 +1,153 @@
+package surrealdb
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+// Client is a wrapper to more easily make HTTP calls to the SurrealDB engine.
+// Any function that accepts a body of type interface{} will do one of two things depending on type.
+// If it is of type string, the body will be the plaintext string, otherwise it will attempt to marshal
+// it into JSON and send that
+type Client struct {
+	// URL is the base URL in SurrealDB to be called
+	URL string
+	// Namespace that you want to connect to
+	NS string
+	// Database that you want to connect to
+	DB string
+	// The user to authenticate as
+	User string
+	// The password to authenticate with
+	Pass string
+}
+
+type Response struct {
+	Time   string      `json:"time"`
+	Status string      `json:"status"`
+	Result interface{} `json:"result"`
+}
+
+// New creates a new instance of a Client
+func NewClient(url, ns, db, user, pass string) Client {
+	return Client{
+		URL:  url,
+		NS:   ns,
+		DB:   db,
+		User: user,
+		Pass: pass,
+	}
+}
+
+// Execute calls the endpoint POST /sql, executing whatever given statement
+func (sc Client) Execute(query string) (Response, error) {
+	return sc.Request("/sql", "POST", query)
+}
+
+// CreateOne calls the endpoint POST /key/:table/:id, executing the statement
+//
+// CREATE type::table($table) CONTENT $body;
+func (sc Client) CreateOne(table, id, body interface{}) (Response, error) {
+	return sc.Request(fmt.Sprintf("/key/%s/%s", table, id), "POST", body)
+}
+
+// CreateAll calls the endpoint POST /key/:table, executing the statement
+//
+// CREATE type::thing($table, $id) CONTENT $body;
+func (sc Client) CreateAll(table string, body interface{}) (Response, error) {
+	return sc.Request(fmt.Sprintf("/key/%s", table), "POST", body)
+}
+
+// SelectAll calls the endpoint GET /key/:table, executing the statement
+//
+// SELECT * FROM type::table($table);
+func (sc Client) SelectAll(table string) (Response, error) {
+	return sc.Request(fmt.Sprintf("/key/%s", table), "GET", "")
+}
+
+// SelectOne calls the endpoint GET /key/:table/:id, executing the statement
+//
+// SELECT * FROM type::thing(:table, :id);
+func (sc Client) SelectOne(table string, id string) (Response, error) {
+	return sc.Request(fmt.Sprintf("/key/%s/%s", table, id), "GET", "")
+}
+
+// ReplaceOne calls the endpoint PUT /key/:table/:id, executing the statement
+//
+// UPDATE type::thing($table, $id) CONTENT $body;
+func (sc Client) ReplaceOne(table, id string, body interface{}) (Response, error) {
```

Comment:
The thing is, this is actually a replace or upsert...

Reply:
Im not sure I understand... Is this code doing the wrong thing, is the comment above incorrect or is the function named incorrectly?

Reply:
Only that the name `Replace` implies that there needs to be something existing before replacing it. However the function actually creates or replaces the record content.

Reply:
I was just using the naming that was in the python lib at https://github.com/surrealdb/surrealdb.py/blob/main/surrealdb/clients/http.py. If we should change it here then we should also change it in that lib too to avoid confusion. The comment about what surrealQL query is taken from the official docs too

Reply:
@tobiemh any updates on this? It'd be nice to get merged so others can improve on it